modules/slowzones/SlowZonesDetails.tsx

```
@@ -1,58 +1,71 @@
 'use client';
 
-import React, { useMemo } from 'react';
+import React from 'react';
 import dayjs from 'dayjs';
-
+import utc from 'dayjs/plugin/utc';
 import { useQuery } from '@tanstack/react-query';
-import { BasicDataWidgetPair } from '../../common/components/widgets/BasicDataWidgetPair';
-import { BasicDataWidgetItem } from '../../common/components/widgets/BasicDataWidgetItem';
+
 import { SlowZonesContainer } from '../../modules/slowzones/SlowZonesContainer';
 import { useDelimitatedRoute } from '../../common/utils/router';
 import { SZWidgetValue, TimeWidgetValue } from '../../common/types/basicWidgets';
-import { useSlowZoneCalculations } from '../../common/utils/slowZoneUtils';
+import { getSlowZoneDeltas } from '../../common/utils/slowZoneUtils';
+import { TODAY_UTC } from '../../common/components/inputs/DateSelection/DateConstants';
 import { fetchAllSlow, fetchDelayTotals } from './api/slowzones';
+dayjs.extend(utc);
 
 export default function SlowZonesDetails() {
   const delayTotals = useQuery(['delayTotals'], fetchDelayTotals);
   const allSlow = useQuery(['allSlow'], fetchAllSlow);
-  const { lineShort } = useDelimitatedRoute();
+  const {
+    lineShort,
+    query: { startDate, endDate },
+  } = useDelimitatedRoute();
 
-  const formattedTotals = useMemo(
-    () =>
-      delayTotals.data && delayTotals.data.filter((t) => new Date(t.date) > new Date(2022, 0, 1)),
-    [delayTotals.data]
-  );
+  const startDateUTC = dayjs.utc(startDate);
+  const endDateUTC = endDate ? dayjs.utc(endDate) : TODAY_UTC;
 
-  const { current, lastWeek, totalDelay, totalDelayLasteek } = useSlowZoneCalculations({
+  const delayTotalsData = delayTotals.data ?? [];
```

if we don't have `allSlow` or `delayTotals` shouldn't we just return an error page? No need to filter over data we don't have